src/psyclone/psyGen.py

```diff
@@ -6957,9 +7143,60 @@ def gen_c_code(self, indent=0):
                                   "children, but found {0}."
                                   "".format(len(self.children)))
 
-        return "(" + self._children[0].gen_c_code() + " " \
-            + self._operator + " " \
-            + self._children[1].gen_c_code() + ")"
+        def operator_format(operator_str,  expr1, expr2):
+            '''
+            :param str operator_str: String representing the operator.
+            :param str expr1: String representation of the LHS operand.
+            :param str expr2: String representation of the RHS operand.
+
+            :returns: C language operator expression.
+            :rtype: str
+            '''
+            return "(" + expr1 + " " + operator_str + " " + expr2 + ")"
+
+        def function_format(function_str, expr1, expr2):
+            '''
+            :param str function_str: Name of the function.
+            :param str expr1: String representation of the first operand.
+            :param str expr2: String representation of the second operand.
+
+            :returns: C language binary function expression.
+            :rtype: str
+            '''
+            return function_str + "(" + expr1 + ", " + expr2 + ")"
+
+        # Define a map with the operator string and the formatter function
+        # associated with each BinaryOperation.Operator
+        opmap = {
+                BinaryOperation.Operator.ADD: ("+", operator_format),
```

Comment:
pylint complains about the indentation of all of these lines...

Reply:
fixed